components/reusable-components/newsletter/emailform/emailform.tsx

```diff
@@ -0,0 +1,66 @@
+'use client';
+
+import React, { useState } from 'react';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+import Notification from './notification/notification';
+import Style from './notification/notification.module.scss';
+
+interface EmailFormProps {
+  inputClassName?: string;
+  buttonClassName?: string;
+}
+
+const EmailForm: React.FC<EmailFormProps> = ({ inputClassName, buttonClassName }) => {
+  const [showNotification, setShowNotification] = useState(false);
+  const [notificationMessage, setNotificationMessage] = useState('');
+  const [notificationType, setNotificationType] = useState<'success' | 'error'>('success');
+
+  const handleSubmit = (values: { email: string }) => {
+    if (values.email) {
+      setNotificationMessage('Subscription successful!');
+      setNotificationType('success');
+      setShowNotification(true);
+    } else {
+      setNotificationMessage('Please enter a valid email address.');
+      setNotificationType('error');
+      setShowNotification(true);
+    }
+  };
+
+  return (
+    <div>
+      {showNotification && (
+        <Notification
+          message={notificationMessage}
+          type={notificationType}
```

Comment:
In this case this is not a bad approach since we only have one input. 
It is better to have the formik initial values, validation schema outside.
We would have const formik = useFormik<InitialValues>({
 <here you would have onSubmit function where you would pass the values>
 and in the component you can do:
`     <form onSubmit={formik.handleSubmit}>`